### PR TITLE
Capturing DB Errors

### PR DESCRIPTION
**Pull request by ankit-il**

### CODE REVIEW

File: controllers/Edd.controller.js [hunk 1]
```diff
@@ -9,13 +9,15 @@ const CartEddFunctions = require("./functions/eddcart_function");
...
 router.get("/edd", async (req, res, next) => {
-   
+   try {
     let cpin = req?.query?.cpin ?? false;
     let skus = req?.query?.skuid ?? false;
     let qty = req?.query?.qty ?? 1;
     let showAll = req?.query?.showAll ?? false;
 
     let funcRes = await EddFunctions.EddMain(cpin,skus,qty);
+
```
Comment (ankit-il) [inline controllers/Edd.controller.js:19]: Remove one newline.

File: controllers/Edd.controller.js [hunk 2]
```diff
@@ -53,11 +55,17 @@ router.get("/edd", async (req, res, next) => {
       }));
     }
     return res.send(response);
+   } catch (error) {
+    // console.log(error);
```
Comment (ankit-il) [inline controllers/Edd.controller.js:59]: Remove the console log and extra line.

File: controllers/Edd.controller.js [hunk 3]
```diff
@@ -53,11 +55,17 @@ router.get("/edd", async (req, res, next) => {
...
+    
+    return res.status(200).send(error);
```
Comment (ankit-il) [inline controllers/Edd.controller.js:61]: Remove the status part.

File: controllers/Edd.controller.js [hunk 4]
```diff
@@ -53,11 +55,17 @@ router.get("/edd", async (req, res, next) => {
...
+   }
 
   });
 
   router.get("/cartedd", async (req, res, next) => {
    
+    try {
     let cpin = req?.query?.cpin ?? false;
```
Comment (ankit-il) [inline controllers/Edd.controller.js:69]: Fix Indentation

File: controllers/Edd.controller.js [hunk 5]
```diff
@@ -53,11 +55,17 @@ router.get("/edd", async (req, res, next) => {
...
+    // console.log(error);
+    
+    return res.status(200).send(error);
+   }
 
   });
 
   router.get("/cartedd", async (req, res, next) => {
    
+    try {
```
Comment (ankit-il) [inline controllers/Edd.controller.js:68]: Fix Indentation

File: controllers/Edd.controller.js [hunk 6]
```diff
@@ -100,7 +108,10 @@ router.get("/edd", async (req, res, next) => {
         deliveryTime:item?.deliveryTime ?? false      
       }));
     }
-    return res.send(response);
+    return res.status(200).send(response);
```
Comment (ankit-il) [inline controllers/Edd.controller.js:111]: Remove status.

File: controllers/functions/Bufferdays/dbd.js [hunk 7]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
 
 module.exports = getDBD;
 async function getDBD(cPin) {
+
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:5]: Remove extra line

File: controllers/functions/Bufferdays/dbd.js [hunk 8]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
 
 module.exports = getDBD;
 async function getDBD(cPin) {
+    
     let promise = new Promise((resolve, reject) => {
         try {
-            console.log("saddasdasdassad");
+            // console.log("saddasdasdassad");
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:8]: Remove this comment.

File: controllers/functions/Bufferdays/dbd.js [hunk 9]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
             promiseEngineConnection.query(
                 `SELECT * FROM DBD_Back_Up WHERE cPin = ${cPin}`,
                 function (error, DBDresults, fields) {
-                    if (error) { console.error(error); }
+                    if (error) { 
+                        console.error(error);
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:13]: Remove this console.

File: controllers/functions/Bufferdays/dbd.js [hunk 10]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
+                        reject({
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:14]:
```
reject({
    "function": "getDBD",
    "responseCode": "410",
    "error": `error fetching data from DBD_Back_Up for cPin: ${cPin}`,
})

File: controllers/functions/Bufferdays/dbd.js [hunk 11]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
+                    
                     if (DBDresults.length) {
-                        console.log('-1-T10-DBD Found');
-                        console.log(JSON.parse(JSON.stringify(DBDresults)));
+                        // console.log('-1-T10-DBD Found');
+                        // console.log(JSON.parse(JSON.stringify(DBDresults)));
+                    
                         resolve(JSON.parse(JSON.stringify(DBDresults)));
                     } else {
                         // console.log('-1-T10-DBD Not Found');
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:35]: Remove all console logs from this function.

File: controllers/functions/Bufferdays/dbd.js [hunk 12]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
-                        resolve([]);
+                        // resolve([]);
+                        reject({
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:37]: Same as https://github.com/SupertailsPCPL/promise-engine/pull/2/files#r1611007419

File: controllers/functions/Bufferdays/dbd.js [hunk 13]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
+                            "responseCode": "411",
+                            "error": "in getting dbdresults length",
+                        })
                     }
                 });
         }
         catch (e) {
-            console.error(e);
-            resolve([]);
+            reject({
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:46]:
```
reject({
    "function": "getDBD",
    "responseCode": "410",
    "errorMessage": `error fetching data from DBD_Back_Up for cPin: ${cPin}`,
    "errorDescription": error,
})

File: controllers/functions/Bufferdays/dbd.js [hunk 14]
```diff
@@ -2,30 +2,52 @@ const promiseEngineConnection = require('../../../dbPromiseEngine');
...
+                        reject({
+                            "getDBD": "",
+                            "responseCode": "410",
+                            "error": "in getting dbd query",
+                        })
+                     }
                     // console.log('-1-T10-DBD');
+                    
                     if (!DBDresults) {
+
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/dbd.js:23]: Reject from here as well.

```
reject({
    "function": "getDBD",
    "responseCode": "410",
    "errorMessage": `No data in DBD_Back_Up for cPin: ${cPin}`,
    "errorDescription": "Not found",
})

File: controllers/functions/Bufferdays/sbd.js [hunk 15]
```diff
@@ -9,21 +9,36 @@ async function getSBD(wareHouseId) {
             promiseEngineConnection.query(
```
Comment (ankit-il) [inline controllers/functions/Bufferdays/sbd.js:9]: Same as controllers/functions/Bufferdays/dbd.js

File: controllers/functions/Cpindata/cpindata.js [hunk 16]
```diff
@@ -8,21 +8,37 @@ async function getcPinData(cPin) {
             promiseEngineConnection.query(
```
Comment (ankit-il) [inline controllers/functions/Cpindata/cpindata.js:8]: Same as controllers/functions/Bufferdays/dbd.js

File: controllers/functions/CutOff/cutoOff.js [hunk 17]
```diff
@@ -8,21 +8,38 @@ async function getCutOff() {
             promiseEngineConnection.query(
```
Comment (ankit-il) [inline controllers/functions/CutOff/cutoOff.js:8]: Same as controllers/functions/Bufferdays/dbd.js

File: controllers/functions/Inventory/inventory.js [hunk 18]
```diff
@@ -24,22 +24,37 @@ async function GetInventory(skuId) {
             promiseEngineConnection.query(
```
Comment (ankit-il) [inline controllers/functions/Inventory/inventory.js:24]: Same as controllers/functions/Bufferdays/dbd.js

Comment (ankit-il) [reply controllers/functions/Inventory/inventory.js:24]: Don't remove the commented console query.

File: controllers/functions/LBD/LBD.js [hunk 19]
```diff
@@ -10,7 +10,11 @@ async function GetLBD(warehouse,cpin,skuWt) {
                 `SELECT * FROM LBD WHERE cpin = ${cpin} AND WH='${warehouse}' AND minWt <= ${skuWt} AND maxWt > ${skuWt};`,
                 async function (error, LBDResults, fields) {
                     if (error) {
-                        console.error(error);
+                        reject({
```
Comment (ankit-il) [inline controllers/functions/LBD/LBD.js:13]: reject({
    "function": "getDBD",
    "responseCode": "410",
    "errorMessage": `error fetching data from LBD for: [${cPin}, ${warehouse} and ${skuWt}`,
    "errorDescription": error,
})

File: controllers/functions/LBD/LBD.js [hunk 20]
```diff
@@ -10,7 +10,11 @@ async function GetLBD(warehouse,cpin,skuWt) {
                 `SELECT * FROM LBD WHERE cpin = ${cpin} AND WH='${warehouse}' AND minWt <= ${skuWt} AND maxWt > ${skuWt};`,
```
Comment (ankit-il) [inline controllers/functions/LBD/LBD.js:10]: Same as controllers/functions/Bufferdays/dbd.js (Change in the errorDescription)

File: controllers/functions/NDD/ndd.js [hunk 21]
```diff
@@ -10,22 +10,37 @@ async function getIsAvailableInNDD(cPin) {
...
+                        reject({
+                            "getIsAvailableInNDD": "inside ndd",
+                            "responseCode": "426",
+                            "error": "in query",
+                        })
                     }
+                    // console.log("NDDREsults")
+                    // console.log(NDDResults);
                     if (NDDResults.length) {
                         dataRes = JSON.parse(JSON.stringify(NDDResults));
```
Comment (ankit-il) [inline controllers/functions/NDD/ndd.js:23]: Add const here.

File: controllers/functions/NDD/ndd.js [hunk 22]
```diff
@@ -10,22 +10,37 @@ async function getIsAvailableInNDD(cPin) {
                 `SELECT * FROM NDD_Pincodes WHERE pincode = "${cPin}" order by priority`,
```
Comment (ankit-il) [inline controllers/functions/NDD/ndd.js:10]: Same as controllers/functions/Bufferdays/dbd.js

File: controllers/functions/Othercouriers/othercourier.js [hunk 23]
```diff
@@ -23,22 +23,37 @@ async function getCourier(cpin, skuWt) {
                 `SELECT * FROM courierV3 WHERE cpin = ${cpin} AND minWt <= ${skuWt} AND maxWt > ${skuWt};`,
                 //`SELECT * FROM courierV2 WHERE rpin = ${cpin} AND WH = '${wareHouseId}' AND minWt <= ${skuWt / 1000} AND maxWt > ${skuWt / 1000} order by EDD;`,
                 async function (error, courierresults, fields) {
-                    if (error) { console.error(error); }
+                    if (error) {
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:26]: Error handling same as dbd. Remove console log from here.

File: controllers/functions/Othercouriers/othercourier.js [hunk 24]
```diff
@@ -23,22 +23,37 @@ async function getCourier(cpin, skuWt) {
...
+                    if (error) { 
+                        console.error(error);
+                        reject({
+                            "getCourier": "inside other courier",
+                            "responseCode": "429",
+                            "error": "error in query of get courier",
+                        })
+                     }
                     if (courierresults) {
                         // console.log('8-T6666 courier');
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:35]: Remove these two console logs.

File: controllers/functions/Othercouriers/othercourier.js [hunk 25]
```diff
@@ -23,22 +23,37 @@ async function getCourier(cpin, skuWt) {
...
                         // console.log(courierresults[0]);
                         resolve(JSON.parse(JSON.stringify(courierresults))[0]);
                     } else {
                         // console.log('-8-T6-courier Not Found');
-                        resolve(false);
+                        reject({
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:40]: Error handling same as dbd. Remove console log from here.

File: controllers/functions/Othercouriers/othercourier.js [hunk 26]
```diff
@@ -23,22 +23,37 @@ async function getCourier(cpin, skuWt) {
...
+                            "getCourier": "inside other courier",
+                            "responseCode": "430",
+                            "error": "error in query of get courier",
+                        })
                     }
                 });
         }
         catch (e) {
             console.error(e);
             // console.log('-8-T6-courier Not Found');
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:50]: Error handling same as dbd. Remove console log from here.

File: controllers/functions/Othercouriers/othercourier.js [hunk 27]
```diff
@@ -52,20 +67,34 @@ async function getWareHousePriority(state) {
             promiseEngineConnection.query(
                 `SELECT * FROM EDDwarehouse_priority WHERE state = ?`, state,
                 async function (error, WarehousePriorityresults, fields) {
-                    if (error) { console.error(error); }
+                    if (error) {
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:70]: Error handling same as dbd.

File: controllers/functions/Othercouriers/othercourier.js [hunk 28]
```diff
@@ -52,20 +67,34 @@ async function getWareHousePriority(state) {
...
+                    if (error) { 
+                        console.error(error);
+                        reject({
+                            "getWareHousePriority": "inside other courier ",
+                            "responseCode": "421",
+                            "error": "in getting  query",
+                        })
+                     }
                     if (WarehousePriorityresults.length) {
                         // console.log('4-T3 wharehouse priority');
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:79]: Remove console log.

File: controllers/functions/Othercouriers/othercourier.js [hunk 29]
```diff
@@ -52,20 +67,34 @@ async function getWareHousePriority(state) {
...
                         // console.log(WarehousePriorityresults);
                         resolve(JSON.parse(JSON.stringify(WarehousePriorityresults))[0]);
                     } else {
                         // console.log('-4-T3-wharehouse Not Found');
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:83]: Error handling same as dbd. Remove console log from here.

File: controllers/functions/Othercouriers/othercourier.js [hunk 30]
```diff
@@ -52,20 +67,34 @@ async function getWareHousePriority(state) {
...
+                            "responseCode": "422",
+                            "error": "in gettin warehousePriority result",
+                        })
                     }
                 });
         }
         catch (e) {
-            console.error(e);
-            resolve(false);
+            reject({
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:93]: Error handling same as dbd.

File: controllers/functions/Othercouriers/othercourier.js [hunk 31]
```diff
@@ -85,19 +114,31 @@ async function otherEDD(cpin, eddResponse) {
     // eddResponse.skuWt = skuWt;
     // console.log("DBD data");
     // console.log(Date());
-   let DBDData = await getDBD(cpin);
+    try {
+        var DBDData=await getDBD(cpin);
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:118]: Space around =

File: controllers/functions/Othercouriers/othercourier.js [hunk 32]
```diff
@@ -85,19 +114,31 @@ async function otherEDD(cpin, eddResponse) {
...
+    } catch (error) {
+        throw error
+    }
+    
+//    let DBDData = await getDBD(cpin);
     const dbdWithNullDate = DBDData?.filter(item => item.date === null).map(item => item.DBD);
     DBD = dbdWithNullDate[0] ?? 0;
     eddResponse = { ...eddResponse, "DBD": `${DBD}` };
     // console.log("Cpin data");
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:127]: Remove console from here.

File: controllers/functions/Othercouriers/othercourier.js [hunk 33]
```diff
@@ -157,9 +204,10 @@ async function otherEDD(cpin, eddResponse) {
         // console.log("Final whareHouseId");
         // console.log(whareHouseId);
     } else {
-        return ({
+
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:207]: Remove extra line.

File: controllers/functions/Othercouriers/othercourier.js [hunk 34]
```diff
@@ -186,7 +234,12 @@ async function otherEDD(cpin, eddResponse) {
     // ware and pincode 
     // console.log("wt ads,ds akads ads kl adsldas da");
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:235]: Remove these two consoles.

File: controllers/functions/Othercouriers/othercourier.js [hunk 35]
```diff
@@ -228,11 +288,21 @@ async function otherEDD(cpin, eddResponse) {
             eddResponse = { ...eddResponse, "EDD": `${EDD}` };
         }
 
-        LBD = await GetLBD(whareHouseId, cpin, skuWt)
+        try {
+            LBD = await GetLBD(whareHouseId, cpin, skuWt)
+        } catch (error) {
+            throw error;
+        }
+        // LBD = await GetLBD(whareHouseId, cpin, skuWt)
```
Comment (ankit-il) [inline controllers/functions/Othercouriers/othercourier.js:296]: Remove this comment.

File: controllers/functions/edd_functions.js [hunk 36]
```diff
@@ -9,45 +9,41 @@ module.exports = { EddMain, getEdd,getCityStateFromPinCode }
...
 //this is the start point of edd - Main Function
 async function EddMain(cpin, skus, qty) {
-    try {
-        if (!skus) {
-            return ({
-                "skuId": "No-Sku-Given",
-                "responseCode": "402",
-                "error": "Enter a SkuId",
-                "errorDiscription": "SERVER ERROR"
+    return new Promise(async(resolve,reject)=>{
```
Comment (ankit-il) [inline controllers/functions/edd_functions.js:12]: Promise not needed. Try/Catch is enough.

File: controllers/functions/edd_functions.js [hunk 37]
```diff
@@ -60,26 +56,30 @@ async function getEdd(cpin, skuId, qty) {
...
             // console.log("GetInventory");
             // console.log(Date());
+        
             let inventoryDetails = await GetInventory(eddResponse.skuId);
+            
+
             // console.log("GetInventory completed");
             // console.log(Date());
             // console.log("aksskakskaa");
             // console.log(inventoryDetails);
```
Comment (ankit-il) [inline controllers/functions/edd_functions.js:66]: Remove the extra lines here.

File: controllers/functions/edd_functions.js [hunk 38]
```diff
@@ -60,26 +56,30 @@ async function getEdd(cpin, skuId, qty) {
             // console.log(eddResponse.skuId);
             // console.log("GetInventory");
             // console.log(Date());
+        
             let inventoryDetails = await GetInventory(eddResponse.skuId);
+
```
Comment (ankit-il) [inline controllers/functions/edd_functions.js:61]: Remove the extra lines here.

File: controllers/functions/edd_functions.js [hunk 39]
```diff
@@ -138,19 +139,32 @@ async function getCityStateFromPinCode(pincode) {
             connection.query(
                 query,
                 async function (error, pincoderesults) {
+
```
Comment (ankit-il) [inline controllers/functions/edd_functions.js:142]: Remove the extra line here.

File: controllers/functions/eddcart_function.js [hunk 40]
```diff
@@ -339,25 +345,30 @@ async function getEdd(cpin, skuId, qty) {
...
                     resolve(b);
                 }
+                
             }
             else {
                 const b = await otherEDD.otherEDD(cpin, eddResponse);
                 resolve(b);
             }
         } catch (e) {
             console.log(e);
```
Comment (ankit-il) [inline controllers/functions/eddcart_function.js:370]: Remove console.

**Review by ankit-il — COMMENTED** [review]
Approved.

**Review by akashsahu1701 — APPROVED** [review]
LGTM

File: controllers/functions/Othercouriers/othercourier.js [hunk 41]
```diff
@@ -112,14 +153,20 @@ async function otherEDD(cpin, eddResponse) {
     // console.log(userState);
     // console.log("warehouse priority");
     // console.log(Date());
-    const wareHousePriorityData = await getWareHousePriority(userState);
+    try {
+        var wareHousePriorityData = await getWareHousePriority(userState);
+    } catch (error) {
+        throw error
+    }
+
```
Comment (akashsahu1701) [inline controllers/functions/Othercouriers/othercourier.js]: remove comments

**Review by akashsahu1701 — APPROVED** [review]

**Pull request merged**